sim/dex-moves.ts

```diff
@@ -448,9 +468,14 @@ export class DataMove extends BasicEffect implements Readonly<BasicEffect & Move
 		this.secondaries = data.secondaries || (this.secondary && [this.secondary]) || null;
 		this.priority = Number(data.priority) || 0;
 		this.category = data.category!;
-		this.defensiveCategory = data.defensiveCategory || undefined;
-		this.useTargetOffensive = !!data.useTargetOffensive;
-		this.useSourceDefensiveAsOffensive = !!data.useSourceDefensiveAsOffensive;
+		this.overrideOffensiveStat =
+			typeof data.overrideOffensiveStat === 'string' ? (data.overrideOffensiveStat as StatIDExceptHP) : undefined;
+		this.overrideOffensivePokemon =
+			data.overrideOffensivePokemon ? (data.overrideOffensivePokemon === "target" ? "target" : "source") : undefined;
+		this.overrideDefensiveStat =
+			typeof data.overrideDefensiveStat === 'string' ? (data.overrideDefensiveStat as StatIDExceptHP) : undefined;
+		this.overrideDefensivePokemon =
+			data.overrideDefensivePokemon ? (data.overrideDefensivePokemon === "source" ? "source" : "target") : undefined;
```

Comment:
```suggestion
		this.overrideOffensiveStat = data.overrideOffensiveStat || undefined;
		this.overrideOffensivePokemon = data.overrideOffensivePokemon || undefined;
		this.overrideDefensiveStat = data.overrideDefensiveStat || undefined;
		this.overrideDefensivePokemon = data.overrideDefensivePokemon || undefined;
```

You're not supposed to have validation here.

Reply:
Didn't think that would work. I had used other possibly undefined strings as references when setting these up. Good to know.